### PR TITLE
python support for intro dp

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: content/4_Gold/Intro_DP.mdx [hunk 1]
```diff
@@ -150,44 +150,46 @@ $\texttt{dp}[1], \texttt{dp}[2], \ldots \texttt{dp}[N]$. Note that in the code
...
 	}
 
-	dp[1] = 0; // base case: 0 cost for the first stone
-
-	for(int i = 1; i <= N; ++i) {
-		if(i + 1 <= N)
+	// dp[1] = 0 is our base case since we're already at the first stone
+	dp[1] = 0;
+	for (int i = 2; i <= N; i++) {
+		dp[i] = INT32_MAX;  // initially set our values to INF
```
Comment (ar492) [inline content/4_Gold/Intro_DP.mdx:177]:
```suggestion
		dp[i] = MAX_N;  // initially set our values to INF
```

File: content/4_Gold/Intro_DP.mdx [hunk 2]
```diff
@@ -150,44 +150,46 @@ $\texttt{dp}[1], \texttt{dp}[2], \ldots \texttt{dp}[N]$. Note that in the code
 we ignore $\texttt{dp}[i]$ if $i>N$.
 
 <LanguageSection>
-
 <CPPSection>
 
 ```cpp
 #include <bits/stdc++.h>
```
Comment (ar492) [inline content/4_Gold/Intro_DP.mdx:157]:
```suggestion
```

**Review by ar492 — CHANGES_REQUESTED** [review]
looks good mostly

File: content/4_Gold/Intro_DP.mdx (hunk 2 above)
Comment (ar492) [inline content/4_Gold/Intro_DP.mdx:157]:
```suggestion
```

**Review by ar492 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Intro_DP.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/4_Gold/Intro_DP.mdx:177]: what

**Review by ar492 — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**